### PR TITLE
Logout user when they click the invite link so the join again to accep…

### PR DESCRIPTION
**Pull request by norkans7**
Fixes https://github.com/rapidpro/rapidpro/issues/1412

### CODE REVIEW

File: templates/orgs/org_join.haml [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
+  .formax
+    .formax-section.open
+      .formax-icon.bg-gray-100.flex.flex-row.items-stretch
+        .margin-wrapper
+          .i-container.icon-user
+      .formax-container.flex-grow
+        .formax-summary
+          -trans "Sign in"
+        .formax-form
+          %form#login-form{method:"post", action:"{% url 'users.user_login' %}?next={% url 'orgs.org_join_accept' secret %}"}
```
Comment (norkans7) [inline templates/orgs/org_join.haml:17]: We show a login form in a formax section and after login we redirect to the join accept invite page

File: temba/orgs/views.py [hunk 2]
```diff
@@ -1965,20 +1966,42 @@ def get_context_data(self, **kwargs):
 
             return context
 
-    class Join(SmartUpdateView):
-        class JoinForm(forms.ModelForm):
+    class Join(SmartTemplateView):
```
Comment (norkans7) [inline temba/orgs/views.py:1969]: The join view here logs out the use which is the link we send in the email

**Comment by rowanseymour** [comment]
Am not quite following why there are so many changes to the join workflow here. Don't we need to just kill the session in the Join view?

**Comment by norkans7** [comment]
I wanted to keep the existing links working and now the join view make sure to log out the authenticated user then once he login back he get to the `join_accept` which has the original functionality

**Comment by nicpottier** [comment]
Wouldn't they still work if we end the session then show the normal view? (asking them to log in or create a new user?)

**Comment by norkans7** [comment]
The only view that is used for authenticated users was the `orgs.org_join` and it same one we redirected to when someone signed is from `orgs.org_create_login`

So we needed the view that will logout the user.
I thought we need to keep the invite link in peoples inbox working so I make `orgs.org_join` do that and the rename the old functionality to `orgs.org_join_accept` and made sure that is only accessible to authenticated users

logging out in the join view as is now will result in loops 

I am no sure what I am missing

**Comment by rowanseymour** [comment]
Want to get this merged so trying to understand what's going on here. The way all this works currently is...

1. User gets an email with a link like https://app.rapidpro.io/org/join/123456/
2. User visits that _Join_ view
   - if they're logged they see a broken view (image below) with a Join button, and if they click that they're added to the org, and redirected to the Inbox.
   - if they're not logged in they get redirected to _Create Login_ where they provide login details, are added to the org, and redirected to the Inbox.

Captura de Pantalla 2021-02-03 a la(s) 09 47 26

I think the fundamental problem there is that you can be logged in with a completely different account to that which the invitation was for. So we shouldn't be looking at `request.user.is_authenticated` to determine whether or not the person needs to create a login. And that seems to be still a problem in this PR.

I think we need to look at the invitation, if that email address exists as a user, then ask them to login, _as that user_. If not, redirect them to the _Create Login_ page.

So maybe we can implement that like:

1. Join page always starts by killing any existing session in `pre_process`.
2. Join page looks up invitation email
   - if email doesn't exist, redirect user to _Create Login_
   - if email does exist, show form with just a password field. If user can authenticate, log them in, redirect to _Inbox_.

@nicpottier thoughts?

**Comment by norkans7** [comment]
So far I assumed we want to allow the user to use the link and join with their existing account even if the email of the account does not match the email on the invite.

On this PR, 
1) user click the link 
2) user gets logged out
3) user enter credentials or if new create new login
4) user granted access by the invite with the account in logged in on step 3)

**Comment by rowanseymour** [comment]
I don't think we should allow that - the logged in user could be a completely different user to the invited user.

**Comment by norkans7** [comment]
Okay,I will update to force to join with the invite email

**Comment by nicpottier** [comment]
Ya, I think Rowan is right here. The big confusion happens when someone clicks on the link while logged in as someone else. I think honestly we should probably simplify this so that you HAVE to accept it as the invited user, you can't accept it as another login. So they either need to create a new account for that user, or ask the inviter to send it to the user account they want to accept it as. This seems to be how much systems like this work anyways on other sites and removes the confusion around having users that weren't invited show up on an account etc..

**Comment by norkans7** [comment]
RapidPro - Visually build nationally scalable mobile applications 2021-02-04 23-22-15
RapidPro - Visually build nationally scalable mobile applications 2021-02-04 23-21-16
RapidPro - Visually build nationally scalable mobile applications 2021-02-04 23-20-37

**Comment by ericnewcomer** [comment]
Agree that if the user exists they stood just get a password field to login but also if they create the login they shouldn't get to fill in the email address either.

File: templates/orgs/org_join.haml [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
+            %fieldset
+              .control-group.hidden
+                %input(type="text" name="username" maxlenght="150" value="{{email}}" readonly="true")
```
Comment (norkans7) [inline templates/orgs/org_join.haml:20]: Only way here is to use a hidden input to pass the email to the login form

**Comment by norkans7** [comment]
RapidPro - Visually build nationally scalable mobile applications 2021-02-05 19-50-13
RapidPro - Visually build nationally scalable mobile applications 2021-02-05 19-49-00

@ericnewcomer @rowanseymour is it necessary to show the email somewhere?

**Comment by nicpottier** [comment]
I think we definitely need to show the email otherwise that is going to be super confusing.

**Comment by norkans7** [comment]
RapidPro - Visually build nationally scalable mobile applications 2021-02-05 21-03-53
 Using the subtitle as email? any other ideas?

**Comment by ericnewcomer** [comment]
I would display it as a field that is either a disabled input or plain text.

**Comment by norkans7** [comment]
Replaced by https://github.com/nyaruka/rapidpro/pull/3180

**Pull request closed without merging**